packages/client/src/execution/preimage.ts

```diff
@@ -0,0 +1,21 @@
+import { DBKey, MetaDBManager } from '../util/metaDBManager'
+
+export class PreimagesManager extends MetaDBManager {
```

Comment:
TBH: I still have my problems to understand what a preimage even is. Can we give this some explicit documentation/explanation? A link to some source also wound't hurt.

Maybe starting here and give the class some line of code docs where an explanation is included.